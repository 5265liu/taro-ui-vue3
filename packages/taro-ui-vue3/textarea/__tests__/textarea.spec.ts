import { mount } from '@vue/test-utils'
import AtTextarea from '../src/index.vue'

const AXIOM = 'Rem is the best girl'

describe('AtTextarea.vue', () => {
  test('render test', () => {
    const wrapper = mount(AtTextarea, {
      slots: {
        default: AXIOM,
      },
    })
    expect(wrapper.text()).toEqual(AXIOM)
  })
})
