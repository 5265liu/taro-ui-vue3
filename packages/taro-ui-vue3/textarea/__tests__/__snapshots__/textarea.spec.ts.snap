// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTextarea Snap render AtTextarea -- props className 1`] = `
<view
  class="at-textarea at-textarea--WEB test"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    0/200
  </view>
</view>
`;

exports[`AtTextarea Snap render AtTextarea -- props count 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
</view>
`;

exports[`AtTextarea Snap render AtTextarea -- props customStyle 1`] = `
<view
  class="at-textarea at-textarea--WEB"
  style="color: red;"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    0/200
  </view>
</view>
`;

exports[`AtTextarea Snap render AtTextarea -- props height 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    0/200
  </view>
</view>
`;

exports[`AtTextarea Snap render AtTextarea -- props maxLength 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="300"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    0/300
  </view>
</view>
`;

exports[`AtTextarea Snap render AtTextarea -- props placeholder 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder="test"
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    0/200
  </view>
</view>
`;

exports[`AtTextarea Snap render AtTextarea -- props value 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    4/200
  </view>
</view>
`;

exports[`AtTextarea Snap render initial AtTextarea 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    class="at-textarea__textarea"
    cursorSpacing="100"
    maxlength="200"
    placeholder=""
    placeholderClass="placeholder"
    selectionEnd="-1"
    selectionStart="-1"
  />
  <view
    class="at-textarea__counter"
  >
    0/200
  </view>
</view>
`;
